Mklinks subprocess -- rcsdiff -c -rR5-Vanilla -rR5-Mvex-r1 server/ddx/sun/RCS/sunInit.c,v 
*** 1.7	1991/09/24 04:54:59
--- 1.8	1991/09/24 05:00:49
***************
*** 53,70 ****
  #include    "dix.h"
  #include    "opaque.h"
  #include    "mipointer.h"
  
  extern int sunMouseProc();
  extern int sunKbdProc();
  extern Bool sunBW2Probe(), sunBW2Create();
  #ifndef MONO_ONLY
! extern Bool sunCG2CProbe(), sunCG2CCreate();
! extern Bool sunCG3CProbe(), sunCG3CCreate();
! extern Bool sunCG4CProbe(), sunCG4CCreate();
! #ifdef FBTYPE_SUNFAST_COLOR /* doesn't exist in sunos3.x */
! extern Bool sunCG6CProbe(), sunCG6CCreate();
  #endif
- #endif
  extern void ProcessInputEvents();
  
  extern void SetInputCheck();
--- 53,67 ----
  #include    "dix.h"
  #include    "opaque.h"
  #include    "mipointer.h"
+ #include    <pixrect/pixrect_hs.h>
  
  extern int sunMouseProc();
  extern int sunKbdProc();
  extern Bool sunBW2Probe(), sunBW2Create();
  #ifndef MONO_ONLY
! extern Bool sunCG8CProbe(), sunCG8CCreate();
! extern Bool sunCG9CProbe();
  #endif
  extern void ProcessInputEvents();
  
  extern void SetInputCheck();
***************
*** 74,83 ****
  #define	XDEVICE	"XDEVICE"
  #define	PARENT	"WINDOW_GFX"
  
  static int autoRepeatHandlersInstalled;	/* FALSE each time InitOutput called */
  
  static Bool sunDevsProbed = FALSE;
! Bool sunSupportsDepth8 = FALSE;
  unsigned long sunGeneration = 0;
  int sunScreenIndex;
  Bool FlipPixels = FALSE;
--- 71,81 ----
  #define	XDEVICE	"XDEVICE"
  #define	PARENT	"WINDOW_GFX"
  
+ Pixrect *sun_pr_screen = NULL;
  static int autoRepeatHandlersInstalled;	/* FALSE each time InitOutput called */
  
  static Bool sunDevsProbed = FALSE;
! Bool sunSupportsDepth24 = FALSE;
  unsigned long sunGeneration = 0;
  int sunScreenIndex;
  Bool FlipPixels = FALSE;
***************
*** 109,122 ****
  sunFbDataRec sunFbData[] = {
  #ifndef MONO_ONLY
      sunBW2Probe,  	"/dev/bwtwo0",	    sunBW2Create,
!     sunCG3CProbe,  	"/dev/cgthree0",    sunCG3CCreate,
! #ifdef sparc
!     sunCG6CProbe,	"/dev/cgsix0",	    sunCG6CCreate,
! #else
!     sunCG2CProbe,  	"/dev/cgtwo0",	    sunCG2CCreate,
!     sunCG4CProbe,  	"/dev/cgfour0",	    sunCG4CCreate,
  #endif
- #endif
      sunBW2Probe,  	"/dev/bwtwo0",	    sunBW2Create,
  };
  
--- 107,115 ----
  sunFbDataRec sunFbData[] = {
  #ifndef MONO_ONLY
      sunBW2Probe,  	"/dev/bwtwo0",	    sunBW2Create,
!     sunCG8CProbe,	"/dev/cgeight0",    sunCG8CCreate,
!     sunCG9CProbe,	"/dev/cgnine0",     sunCG8CCreate,
  #endif
      sunBW2Probe,  	"/dev/bwtwo0",	    sunBW2Create,
  };
  
***************
*** 139,145 ****
  
  static PixmapFormatRec	formats[] = {
      1, 1, BITMAP_SCANLINE_PAD,	/* 1-bit deep */
!     8, 8, BITMAP_SCANLINE_PAD,	/* 8-bit deep */
  };
  #define NUMFORMATS	(sizeof formats)/(sizeof formats[0])
  
--- 132,138 ----
  
  static PixmapFormatRec	formats[] = {
      1, 1, BITMAP_SCANLINE_PAD,	/* 1-bit deep */
!    24,32, BITMAP_SCANLINE_PAD,	/*24-bit deep */
  };
  #define NUMFORMATS	(sizeof formats)/(sizeof formats[0])
  
***************
*** 244,253 ****
  		sunFbData[dev].createProc = NULL;
  	}
  	sunDevsProbed = TRUE;
  	if (n == 0)
  	    return;
      }
!     if (!sunSupportsDepth8)
  	pScreenInfo->numPixmapFormats--;
      for (i = NUMSCREENS, dev = devStart; --i >= DEV_END; dev++) {
  	if (sunFbData[dev].createProc)
--- 237,247 ----
  		sunFbData[dev].createProc = NULL;
  	}
  	sunDevsProbed = TRUE;
+ 
  	if (n == 0)
  	    return;
      }
!     if (!sunSupportsDepth24)
  	pScreenInfo->numPixmapFormats--;
      for (i = NUMSCREENS, dev = devStart; --i >= DEV_END; dev++) {
  	if (sunFbData[dev].createProc)
***************
*** 444,449 ****
--- 438,445 ----
  	while (*dList && *dList != ':') {
  	    dList++;
  	}
+ 	if (*dList == ':')
+ 	    dList++;
      }
      if (*dList) {
  	register char *cp = dList;
***************
*** 765,770 ****
--- 761,772 ----
  	    (void) close(fd);
  	    return (-1);
  	}
+     }
+ 
+     if (!sun_pr_screen)
+     {
+ 	sun_pr_screen = pr_open("/dev/fb");
+ 	pr_set_plane_group(sun_pr_screen, PIXPG_24BIT_COLOR);
      }
  
      if (name && strcmp (name, "/dev/fb") == 0) {
